.github/workflows/devdocs.yml

```diff
@@ -0,0 +1,68 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Upload dev docs
+
+on:
+  push:
+  pull_request:
```

Comment:
Do we need these events?

Reply:
No, that's just temporary for testing on this PR. After that only the cron job will be sufficient

Reply:
OK.
Could you remove `push` and `pull_request` before we merge this?

We can keep `pull_request` if we use the forked repository's `gh-pages` branch instead of `apache/arrow-site`'s `asf-site` branch like our existing `.github/workflows/deploy.yml` does.

Reply:
Yes, will certainly update this before merging

Reply:
> We can keep `pull_request` if we use the forked repository's `gh-pages` branch instead of `apache/arrow-site`'s `asf-site` branch like our existing `.github/workflows/deploy.yml` does.

OK, I see how this is done in deploy.yml. Now, for the docs I think that's probably less relevant, as it is not content from a PR that is being built (as is the case for the main site here), but rather just fetched from elsewhere. But we can always add it later if it would be useful to have.